File: create-credential-directly.js
#!/usr/bin/env node
/**
 * Create credential directly using SignifyTS
 */

const { SignifyClient, Tier, ready } = require('signify-ts');

async function createCredentialDirectly() {
  console.log('Creating credential directly using SignifyTS...');
  
  try {
    await ready();
    
    // Use the issuer bran from the brans.json file
    const bran = 'BG4MS-QY9f__GsTghCQZA';
    console.log('Using issuer bran:', bran);
    
    // Create client
    const client = new SignifyClient('http://localhost:3905', bran, Tier.low, 'http://localhost:3906');
    
    // Try to connect (this might fail, but let's see)
    console.log('Attempting to connect...');
    try {
      await client.connect();
      console.log('✅ Connected successfully!');
    } catch (connectError) {
      console.log('⚠️ Connection failed, but continuing...');
      console.log('Connection error:', connectError.message);
    }
    
    // Try to get existing identifiers
    console.log('Getting identifiers...');
    try {
      const identifiers = await client.identifiers().list();
      console.log('Found identifiers:', identifiers.length);
      
      for (const identifier of identifiers) {
        console.log(`  - ${identifier.name}: ${identifier.prefix}`);
      }
      
      // If we have identifiers, try to get registries
      if (identifiers.length > 0) {
        console.log('Getting registries...');
        const registries = await client.registries().list(identifiers[0].name);
        console.log('Found registries:', registries.length);
        
        for (const registry of registries) {
          console.log(`  - ${registry.name}: ${registry.regk}`);
        }
        
        // Try to issue a credential
        console.log('Attempting to issue credential...');
        try {
          const credentialData = {
            ri: registries[0].regk, // Use first registry
            s: "Etk6EhuG09iXynnvGIxUZxYwugDCNC9G80ouiJ1apMlU", // Travel preferences schema
            a: {
              employeeId: "EMP-DIRECT-001",
              seatPreference: "window",
              mealPreference: "vegetarian",
              airlines: "SAS,Lufthansa",
              emergencyContact: "Emergency Contact +46701234567",
              allergies: "nuts,shellfish"
            }
          };
          
          console.log('Credential data:', JSON.stringify(credentialData, null, 2));
          
          const result = await client.credentials().issue(identifiers[0].name, credentialData);
          console.log('✅ Credential issued successfully!');
          console.log('Credential SAID:', result.acdc?.ked?.d);
          
          return result.acdc?.ked?.d;
        } catch (issueError) {
          console.log('❌ Credential issuance failed:', issueError.message);
          console.log('Issue error details:', issueError);
        }
      } else {
        console.log('No identifiers found');
      }
    } catch (identifiersError) {
      console.log('❌ Identifiers operation failed:', identifiersError.message);
    }
    
    return null;
    
  } catch (error) {
    console.log('❌ Operation failed:', error.message);
    console.log('Error details:', error);
    return null;
  }
}

// Run the credential creation
createCredentialDirectly().then(credentialSaid => {
  if (credentialSaid) {
    console.log('\n🎉 Credential successfully created with SAID:', credentialSaid);
  } else {
    console.log('\n💥 Credential creation failed!');
  }
});

File: create-credential-directly.py
#!/usr/bin/env python3
"""
Create a credential directly using KERIA API
"""

import requests
import json
import time

def create_credential_directly():
    """Create a credential directly using KERIA API"""
    
    print("Creating credential directly via KERIA API...")
    
    # KERIA endpoints
    keria_base = "http://localhost:3905"
    
    # First, let's try to boot an agent
    print("1. Booting agent...")
    boot_data = {
        "name": "direct-issuer",
        "passcode": "direct-passcode-123"
    }
    
    try:
        response = requests.post(f"{keria_base}/boot", json=boot_data)
        print(f"   Boot response: {response.status_code}")
        if response.status_code in [200, 201]:
            print("   Agent booted successfully")
        elif response.status_code == 409:
            print("   Agent already exists")
        else:
            print(f"   Boot failed: {response.text}")
    except Exception as e:
        print(f"   Boot error: {e}")
    
    # Try to connect to the agent
    print("2. Connecting to agent...")
    try:
        # This would normally require authentication headers
        # But let's first see what endpoints are available
        response = requests.get(f"{keria_base}/")
        print(f"   KERIA root response: {response.status_code}")
        print(f"   Response: {response.text[:200]}")
    except Exception as e:
        print(f"   Connection error: {e}")
    
    # Try to get identifiers
    print("3. Getting identifiers...")
    try:
        response = requests.get(f"{keria_base}/identifiers")
        print(f"   Identifiers response: {response.status_code}")
        if response.status_code == 200:
            identifiers = response.json()
            print(f"   Found {len(identifiers)} identifiers")
            for identifier in identifiers:
                print(f"     - {identifier.get('name', 'Unknown')}: {identifier.get('prefix', 'Unknown')}")
        else:
            print(f"   Failed to get identifiers: {response.text}")
    except Exception as e:
        print(f"   Identifiers error: {e}")
    
    print("\nDirect KERIA API approach completed.")

if __name__ == "__main__":
    create_credential_directly()